core/src/saros/communication/info/InfoManager.java

```diff
@@ -0,0 +1,182 @@
+import saros.net.xmpp.contact.XMPPContact;
+import saros.net.xmpp.contact.XMPPContactsService;
+
+/**
+ * Component for figuring out whether two Saros plug-in instances with known Version are compatible
+ * and to exchange Infos outside of Sessions.
+ *
+ * <p>This class compares if local and remote version (not checking qualifier) are the same.
+ *
+ * <p>If you want to implement backward compatibility in a later version, as a suggestion:
+ * acknowledge same major.minor version as compatible. Alternatively add a `backwards_compatibility`
+ * key with the last working version as value and add a check for it.
+ */
+@Component(module = "core")
+public class InfoManager {
+  private static final Logger log = Logger.getLogger(InfoManager.class);
+
+  private final Version localVersion;
+  private final ITransmitter transmitter;
+  private final XMPPContactsService contactsService;
+
+  private final ConcurrentHashMap<String, String> localInfo = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<XMPPContact, ClientInfo> remoteInfo = new ConcurrentHashMap<>();
+
+  private final PacketListener infoListener =
+      packet -> {
+        log.debug("received info from " + packet.getFrom());
+        handleInfo(packet);
+      };
+
+  private final IContactsUpdate contactsUpdateListener =
+      (contact, updateType) -> {
+        if (updateType == UpdateType.STATUS) {
+          if (contact.map(c -> c.getStatus().isOnline()).orElse(false)) sendInfo(contact.get());
+          else contact.ifPresent(remoteInfo::remove);
+        } else if (updateType == UpdateType.REMOVED) {
+          contact.ifPresent(remoteInfo::remove);
+        } else if (updateType == UpdateType.NOT_CONNECTED) {
+          remoteInfo.clear();
+        }
+      };
+
+  public InfoManager(
+      @SarosVersion String version,
+      IReceiver receiver,
+      ITransmitter transmitter,
+      XMPPContactsService contactsService) {
+    this.localVersion = Version.parseVersion(version);
+    if (this.localVersion == Version.INVALID)
+      throw new IllegalArgumentException("version string is malformed: " + version);
+
+    localInfo.put(ClientInfo.VERSION_KEY, localVersion.toString());
+
+    this.transmitter = transmitter;
+    this.contactsService = contactsService;
+    contactsService.addListener(contactsUpdateListener);
+
+    receiver.addPacketListener(
+        infoListener,
+        new AndFilter(
+            InfoExchangeExtension.PROVIDER.getIQFilter(),
+            packet -> ((IQ) packet).getType() == IQ.Type.SET));
+  }
+
+  /**
+   * Determines the version compatibility with the given peer.
+   *
+   * @param rqJID the resource qualified JID of the peer
+   * @return the {@link VersionCompatibilityResult VersionCompatibilityResult}
+   */
+  public VersionCompatibilityResult determineVersionCompatibility(JID rqJID) {
+    Version remoteVersion =
+        contactsService
+            .getContact(rqJID.getRAW())
```

Comment:
It doesn't seem to matter if you use a FQ JID or not. I guess you could also use the FQJID of an instant messager and would get the Saros version as long as the user is also connected with a Saros client.

Reply:
yes, changed to jid